### ISSUE TITLE
[Faceted Filters] Mobile close control missing accessible name

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#117]
## Issue summary

When the Facet window is open while using a mobile screen reader, the Close control is missing its accessible name/alternative text. When focused, screen readers will announce the control type and state only. This may create a confusing user experience when the control purpose os not provided.

<details>
  <summary>Screenshots</summary>
<img width="493" alt="Screen shot of mobile faceted filters section. Close control is highlighted." src="https://user-images.githubusercontent.com/1392632/124609054-46f89680-de3d-11eb-8bb4-3882159c405d.png">

</details>

#### HTML

```html
<span tabindex="0" class="mobile-facets__close mobile-facets__close--no-js">
  <!-- SVG… -->
</span>
```

## Steps to reproduce

1. Go to a collection page with Dawn theme with any supported mobile browser.
2. Activate the "Filter and Sort" control.
3. With an active screen reader, navigate to the Close control.

### Expected

- Controls to feature an accessible name for assistive technology.

### Actual

- Close control is missing its accessible name.

## Recommendation

Adjust the CSS to _not_ hide the `.mobile-facets__open` container which houses the control accessible name.

#### CSS

```css
details[open] .mobile-facets__open {
  /* visibility: hidden; */
}
```

<!-- #### Diff

```diff

``` -->

## Specifications

- **Component:** Faceted Filters
- **WCAG Principal:** Operable
- **WCAG SC:** [1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- **Severity:** High
- **Effort:** Low
<!-- - **Related:** -->
